tests/cover/test_fspaths.py

```diff
@@ -0,0 +1,254 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import io
+import os
+import sys
+import codecs
+import tempfile
+
+import pytest
+
+from hypothesis import find, given
+from hypothesis.strategies import fspaths, from_type
+from hypothesis.internal.compat import PY3, text_type
+
+try:
+    import pathlib
+except ImportError:
+    pathlib = None
+
+
+encoding = sys.getfilesystemencoding()
+is_win = (os.name == 'nt')
+
+
+def norm_encoding(name):
+    """Normalizes an encoding name."""
+
+    return codecs.lookup(name).name
+
+
+def single_byte_full_encoding(encoding):
+    """Whether the encoding can decode all byte values (e.g. latin1)"""
+
+    if PY3:
+        bytes_ = map(lambda i: bytes([i]), range(0, 256))
+    else:
+        bytes_ = map(chr, range(0, 256))
+
+    for byte in bytes_:
+        try:
+            byte.decode(encoding)
+        except UnicodeDecodeError:
+            return False
+    return True
+
+
+def fspath(p):
+    """Like os.fspath but for Python <= 3.6."""
+
+    try:
+        return os.fspath(p)
+    except AttributeError:
+        return p
+
+
+@pytest.fixture(scope='module')
+def tempdir_path():
```

Comment:
👍, but I'd delete the `print` and use `try: yield dir_; finally: os.rmdir(dir_)` to ensure the cleanup always happens